client/src/panels/panel_declarations/services/services_channels.js

```diff
@@ -0,0 +1,164 @@
+import text from './services.yaml';
+
+import { Service } from '../../../models/services.js';
+
+import {
+  create_text_maker_component,
+  declare_panel,
+  InfographicPanel,
+  NivoResponsiveBar,
+  declarative_charts,
+  util_components,
+} from "../shared.js";
+
+const {
+  Select,
+} = util_components;
+
+const { GraphLegend } = declarative_charts;
+import { Fragment } from 'react';
+
+const { text_maker, TM } = create_text_maker_component(text);
+
+
+
+
+class ServicesChannelsPanel extends React.Component {
+  constructor(){
+    super();
+    this.state = {
+      selected_service_id: "all",
+    };
+  }
+  
+  render(){
+    const {
+      panel_args,
+    } = this.props;
+    const { selected_service_id } = this.state;
+    const colors = infobase_colors();
+
+    const data_keys = ["telephone_enquires",	"website_visits",	"online_applications",	"in_person_applications",	"mail_applications",	"other_channel_applications"];
+
+    const all_services = _.chain({})
+      .mergeWith(..._.map(panel_args.services, serv=>_.pick(serv,data_keys)),_.add)
+      .merge({name: text_maker("all"), id: "all"})
+      .value();
+
+    const selected_service = selected_service_id === "all" ? 
+      all_services :
+      _.find(panel_args.services, serv => serv.id === selected_service_id) || _.first(panel_args.services);
+    
+    const selected_title = selected_service.name;
+
+    const bar_data = _.chain(data_keys)
+      .map(key => ({
+        number: selected_service[key],
+        label: text_maker(`label_${key}`),
+      }) )
+      .each(obj => {obj[text_maker("number_of_enquiries")] = obj.number;})
+      .value();
+
+    const options = _.chain([all_services])
+      .concat(panel_args.services)
+      .map(({name, id}) => ({
+        id,
+        display: name,
+      }) )
+      .value();
+    
+    const all_feedback_channels = ["Online", "Postal Mail", "Fax", "Email", "In-Person", "Telephone"];
```

Comment:
Might be useful to put these, and the `data_keys` too, somewhere like business constants, on as properties of the Services subject? They feel like something useful to have a single source of truth for. Check me on that, I'm not as close to this stuff as you at the moment.

Reply:
A lot of the services data has constants like these that are relevant for a single aspect of the services (in this case they're the possible values of the available feedback channels) but nowhere else. I think there's more of these than exist for the other data we've got.

Probably `client/src/models/services.js` is the right spot for them, but fair warning there's going to be a bunch...